src/ooxml/java/org/apache/poi/xssf/usermodel/helpers/XSSFRowShifter.java

```diff
@@ -60,228 +48,130 @@ Licensed to the Apache Software Foundation (ASF) under one or more
-                String[] regions = stRef.toString().split(" ");
-                for (String region : regions) {
-                    cellRanges.add(CellRangeAddress.valueOf(region));
-                }
-            }
-
-            boolean changed = false;
-            List<CellRangeAddress> temp = new ArrayList<>();
-            for (CellRangeAddress craOld : cellRanges) {
-                CellRangeAddress craNew = shiftRange(shifter, craOld, sheetIndex);
-                if (craNew == null) {
-                    changed = true;
-                    continue;
-                }
-                temp.add(craNew);
-                if (craNew != craOld) {
-                    changed = true;
-                }
-            }
-
-            if (changed) {
-                int nRanges = temp.size();
-                if (nRanges == 0) {
-                    ctWorksheet.removeConditionalFormatting(j);
-                    continue;
-                }
-                List<String> refs = new ArrayList<>();
-                for(CellRangeAddress a : temp) refs.add(a.formatAsString());
-                cf.setSqref(refs);
-            }
-
-            for(CTCfRule cfRule : cf.getCfRuleArray()){
-                String[] formulaArray = cfRule.getFormulaArray();
-                for (int i = 0; i < formulaArray.length; i++) {
-                    String formula = formulaArray[i];
-                    Ptg[] ptgs = FormulaParser.parse(formula, fpb, FormulaType.CELL, sheetIndex, rowIndex);
-                    if (shifter.adjustFormula(ptgs, sheetIndex)) {
-                        String shiftedFmla = FormulaRenderer.toFormulaString(fpb, ptgs);
-                        cfRule.setFormulaArray(i, shiftedFmla);
-                    }
-                }
-            }
-        }
+    public List<CellRangeAddress> shiftMerged(int startRow, int endRow, int n) {
+        return shiftMergedRegions(startRow, endRow, n);
     }
     
     /**
-     * Shift the Hyperlink anchors (not the hyperlink text, even if the hyperlink
-     * is of type LINK_DOCUMENT and refers to a cell that was shifted). Hyperlinks
-     * do not track the content they point to.
-     *
-     * @param shifter
-     */
+        @deprecated, use FormulaShiftingManager.updateNamedRanges() directly instead
+    */
+    public void updateNamedRanges(FormulaShifter shifter) {
+        formulaShiftingManager.updateNamedRanges();
+    }
+    /**
+        @deprecated, use FormulaShiftingManager.updateFormulas() directly instead
+    */ 
+    public void updateFormulas(FormulaShifter shifter) {
+        formulaShiftingManager.updateFormulas();
+    }
+    /**
+        @deprecated, use FormulaShiftingManager.updateConditionalFormatting() directly instead
+     */ 
+    public void updateConditionalFormatting(FormulaShifter shifter) {
+        formulaShiftingManager.updateConditionalFormatting();
+    }    
+    /**
+        @deprecated, use FormulaShiftingManager.updateHyperlinks() directly instead
```

Comment:
could you include the 4.0.0 version number in the deprecation messages?

Reply:
Ok. I have also added @Deprecated annotations. Now it looks like 

/**
        @deprecated in POI 4.0.0, use FormulaShiftingManager.updateHyperlinks() directly instead
     */ 
    @Deprecated
    public void updateHyperlinks(FormulaShifter shifter) {

Is that fine ?

Reply:
The javadoc `@deprecated` is lower case.
The Java Annotation `@Deprecated` is title case.